src/include/rear/ui.rb

```diff
@@ -441,25 +527,78 @@ def RearConfigDialog
           "<p><b>OK</b> saves the configuration and quits while <b>Cancel</b> closes the configuration dialog without saving.<p>"
         )
 
-
       # get varibales from config
       netfs_url = [Rear.netfs_url]
       netfs_keep_old_backup = Rear.netfs_keep_old_backup
-      modules_load = deep_copy(Rear.modules_load)
-      backup_prog_include = deep_copy(Rear.backup_prog_include)
+      use_dhclient = Rear.use_dhclient
+      modules_load = Rear.modules_load
+      backup_prog_include = Rear.backup_prog_include || []
+      post_recovery_script = Rear.post_recovery_script || []
+      required_progs = Rear.required_progs || []
+      copy_as_is = Rear.copy_as_is || []
+      backup_options = Rear.backup_options
       output = Rear.output
 
+      # set defaults:
+      if RearSystemCheck.Btrfs? && backup_prog_include.empty?
+        backup_prog_include = GetMountpoints()
+      end
+
+      # Fixme: Check item by item 
+      if required_progs.empty?
+        required_progs = %w(snapper chattr lsattr)
+      end
+
+      # Fixme: Check item by item
+      if copy_as_is.empty?
+        copy_as_is = %w(/usr/lib/snapper/installation-helper /etc/snapper/config-templates/default)
+      end
+
+      if backup_options.empty?
+        backup_options = "nfsvers=3,nolock"
+      end
+
+      # Maybe fixme: It is too complex to find out if this line is maybe already there.
+      # but maybe we can squeeze this into a shellscript and include this in the rear package?
+      # I even think we can run the 'if'-part already here and either include the 'then' part or not.
+      # To be clearified: Huha says quotas are broken. So ask jsmeix why he included this here.
+      if post_recovery_script.empty?
+        post_recovery_script = [
+          'if snapper --no-dbus -r $TARGET_FS_ROOT get-config | grep -q "^QGROUP.*[0-9]/[0-9]" ; '\
+          'then snapper --no-dbus -r $TARGET_FS_ROOT set-config QGROUP= ;'\
+          ' snapper --no-dbus -r $TARGET_FS_ROOT setup-quota && echo snapper setup-quota done'\
+          ' || echo snapper setup-quota failed ; '\
+          'else echo snapper setup-quota not used ; '\
+          'fi'
+        ]
```

Comment:
so script is single element array with one string? and I agree it is quite complex shell script. I worry it can be hard to debug problems in it on customer side.